source/interprocedural_analyses/taint/rule.ml

```diff
@@ -18,6 +18,9 @@ type t = {
   code: int;
   name: string;
   message_format: string; (* format *)
+  (* taint config location information if exists*)
+  taint_config_path: PyrePath.t option;
+  taint_config_location: JsonParsing.Location.t option;
```

Comment:
Instead of storing it this way, what you could do is have a `Json.Location.WithPath.t` or `Json.LocationWithPath.t`
Also, why can it be `None`?

Reply:
Yes, `LocationWithPath.t` is better.

> Also, why can it be None?

We define some rules implicitly in the code